### DISCUSSION TITLE
Zigbee2mqtt & Internet instance of HA

### DISCUSSION THREAD

**Discussion by adi90x** [Koenkk/zigbee2mqtt#12522] [General]
Hello,

Quick question, I am looking to move my HA instance to a different server,  in this idea I want to split HA & zigbee coordinator thus moving to zigbee2mqtt, used to use ZHA.
Quick question, if installing zigbee2mqtt on a pi with the zigbee coordinator  and installing HA instance and mqtt server on another server, what happen if internet connection drop between those two ? Are all zigbee device update lost ? Or are they cache on zigbee2mqtt side until internet is restored ?

Regards,
Adrien

**Comment by alanmgomes** [comment]
They will report as unavailable devices on HA if zigbee2mqtt or mqtt server is down or unable to be reached. 
U can test the behavior by shutting down zigbee2mqtt only:

![image](https://user-images.githubusercontent.com/42685254/168901892-b4fcafd1-d063-4758-8770-31be42b7e2da.png)

![image](https://user-images.githubusercontent.com/42685254/168902357-4422675d-21a1-465c-b324-31c86c92341f.png)

**Comment by adi90x** [comment]
Hey, thx for the answer, make sense device are unavailble.
My question was more if everything run fine on both side but mqtt server can't be reach anymore for some time (internet down), will there be a cache on zigbee2mqtt side that will be push to mqtt server when internet is up again ?

**Reply by alanmgomes** [reply]
> Hey, thx for the answer, make sense device are unavailble. My question was more if everything run fine on both side but mqtt server can't be reach anymore for some time (internet down), will there be a cache on zigbee2mqtt side that will be push to mqtt server when internet is up again ?

No, all communications are synchronous, so if one device report something to zigbee2mqtt, immediately a message will be posted on a topic.

After the communication come back online, all new status reported by devices will be updated, but the old ones are lost. In mean time, u can see in the zigbee2mqtt logs a plenty of errors like this:

![image](https://user-images.githubusercontent.com/42685254/168905077-15d49e57-68b0-4e08-ac98-72834349c290.png)

After the communication come back online everything should work:

![image](https://user-images.githubusercontent.com/42685254/168905921-b39b1da3-772b-4ee0-bc98-0f0621925696.png)

**Comment by adi90x** [comment]
But for example if I have a temperature sensor, if it publish normally one
message every min, if at some point internet is down for an hour, will
there be 60 measures pushed to mqtt server when everything resume ?

**Comment by alanmgomes** [comment]
No, all measures and whatweaver other info sent by any device will be lost in meantime
